sunpy/net/dataretriever/client.py

```diff
@@ -336,6 +337,15 @@ def fetch(self, qres, path=None, error_callback=None, **kwargs):
         -------
         Results Object
         """
+        # Check for type of path
+        if isinstance(path, str) or path is None:
+            pass
```

Comment:
Do we need this if statement?

Reply:
Also I wonder why the documentation string for this is  empty of the keywords.

Reply:
This is to ensure that the path which is `None`, `str` or `pathlib.Path` goes past. Any other type would `raise` a `TypeError`

Reply:
Could the `if,elif,else` block be refactored to avoid having the if:pass part?

Reply:
I don't see how we can go past not checking whether path is `str` or `None`.

If you have some suggestion on how to do this, can you share?

Reply:
@Cadair might

Reply:
I haven't tested this, so it probably has several syntax issues, but something like this maybe:

```
if path is not None:
    if isinstance(path, pathlib.Path) and HAS_PATHLIB:
        path = str(path.absolute())
    elif not isinstance(path, str): # need py2 fix for six here?
        raise TypeError("path keyword is invalid. Should be a 'str' or 'pathlib.Path'.\
                         Got '{}'.".format(type(path)))
``` 

I think if we check for str we need to use six to ensure python 2 compatibility.